Add a redis image managed by s6

sandytrinh opened the pull request:
Hi @smebberson 

This is ready for review!

sandytrinh committed:
Added a redis image with redis managed by s6

sandytrinh committed:
Added local custom redis configuration file which can be replace in downstream images. Replace wget with curl in dockerfile as wget is not installed. Added fix to remove hugepage warning

smebberson commented:
@sandytrinh, I think we should run `redis-server` as a new `redis` unprivileged user created in the docker file.

sandytrinh committed:
Updated redis image so that the redis serve is run by an unprivileged user 'redis'

sandytrinh commented:
Hi @smebberson 

I've updated the redis image so the redis server is run by an unprivileged user 'redis'.
I've also added ENV variable for the redis version in the `Dockerfile` so we can easily version bump redis when needed.

Please review!

sandytrinh committed:
Updated redis image to stop it's host container if the redis service dies

sandytrinh commented:
Also updated s6 to stop the container if redis dies.
